dspace-spring-rest/src/main/java/org/dspace/app/rest/model/SearchConfigurationRest.java

```diff
@@ -104,15 +104,50 @@ public int hashCode() {
 
     public static class Filter {
         private String filter;
+        private boolean hasFacets = false;
+        private String type;
+        private boolean isOpenByDefault = false;
         private List<Operator> operators = new LinkedList<>();
+        private int pageSize;
 
         public static final String OPERATOR_EQUALS = "equals";
         public static final String OPERATOR_NOTEQUALS = "notequals";
         public static final String OPERATOR_AUTHORITY = "authority";
         public static final String OPERATOR_NOTAUTHORITY = "notauthority";
         public static final String OPERATOR_CONTAINS = "contains";
         public static final String OPERATOR_NOTCONTAINS = "notcontains";
+        public static final String OPERATOR_QUERY = "query";
 
+        public boolean isHasFacets() {
+            return hasFacets;
+        }
+        public void setHasFacets(boolean hasFacets) {
+            this.hasFacets = hasFacets;
+        }
+
+        public int getPageSize() {
+            return pageSize;
+        }
+
+        public void setPageSize(int pageSize) {
+            this.pageSize = pageSize;
+        }
+
+        public String getType() {
```

Comment:
Several new methods here that are not always self-explanatory. Again, they need Javadocs.

Reply:
These fields are also undocumented here: https://github.com/DSpace/DSpace/blob/dspace-6.3/dspace-api/src/main/java/org/dspace/discovery/configuration/DiscoverySearchFilter.java

In order to prevent adding useless documentation in the form of `The type of the search filter`: Do you expect us to enumerate and explain the different types of filters here?

Do you want us to retroactively document the API classes as well in this PR?

Reply:
You don't need to retroactively document API classes.  However, new PRs should add Javadocs for all Public Classes and Methods. We currently do not "fail" PRs that do not provide Javadocs, however this goal is listed in our Code Style (see #6): https://wiki.duraspace.org/display/DSPACE/Code+Style+Guide#CodeStyleGuide-DSpaceJavaStyleGuide(Adopted2018,forDSpace7.xandabove)

When I review PRs, I make it an effort to check for JavaDocs.  I expect others to do so as well, since it's in our Code Style.  However, I'm sure there are PRs that have slipped through with less than perfect JavaDocs.

As these are new public methods that do not have `@Override`, if we run JavaDocs, they will be entirely undocumented.  Even a simple comment like `The type of search filter` *does* provide more information -- `getType` is not very specific.  We use `type` to represent a lot of things in DSpace.

Reply:
The particular setType() that I just looked at (https://github.com/atmire/DSpace/blob/91218ed36c5fa04d3b2097cb373c06e66b82968f/dspace-spring-rest/src/main/java/org/dspace/app/rest/model/SearchConfigurationRest.java#L140) is a very nice example.  It takes a String argument.  We have no way of knowing, from outside the class, what are the consequences of setting a "type", because we don't know what "type" means and cannot infer anything from the nature of the argument.  Should I encounter a reference to this method while trying to read some other code, when I drop the pointer on it and look in my IDE's Javadoc window, it only tells me words to the effect that the author didn't bother to explain what it means to call this method.  I will have to turn away from my task to spend several minutes (occasionally hours!) to comprehend code I wasn't working on.  I'll probably capture my hard-won understanding by documenting the mystery method myself -- perhaps even correctly.  It would be better to know what the author was thinking.

I would agree that "the type of the search filter" is useless documentation, because we still don't know what that means.  What does it mean for a filter to have one type rather than another?  Is this an arbitrary label (and, if so, what uses this label?), or do I need to go dig through still more code to find the accepted values and what *they* mean?  "The type of the search filter" is the sort of comment I would write when I've been studying this class for 20-30 minutes, together with its friends and relations, and I still don't understand what "type" means.  Things like that get written because some tool is failing the build until something is written but I have no idea what to write.